schema/any.go

```diff
@@ -69,16 +61,152 @@ func (a *AnySchema) ValidateCompatibility(typeOrData any) error {
 	default:
 		// Schema is not a primitive, slice, or map type, so check the complex types
 		// Explicitly allow object schemas since their reflected type can be a struct if they are struct mapped.
-		switch typeOrData.(type) {
+		switch schema.(type) {
 		case *AnySchema, *OneOfSchema[int64], *OneOfSchema[string], *ObjectSchema:
 			// These are the allowed values.
 		default:
 			// It's not an any schema or a type compatible with an any schema, so error
 			return &ConstraintError{
-				Message: fmt.Sprintf("schema type `%T` cannot be used as an input for an 'any' type", typeOrData),
+				Message: fmt.Sprintf("schema type `%T` cannot be used as an input for an 'any' type", schema),
+			}
+		}
+		return nil
+	}
+}
+
+func (a *AnySchema) validateAnyMap(data map[any]any) error {
+	// Test individual values
+	var firstReflectKind reflect.Kind
+	for key, value := range data {
+		// Validate key
+		reflectKind := reflect.ValueOf(key).Kind()
+		switch reflectKind {
+		// While it is possible to add more types of ints, it's likely better to keep it consistent with i64
+		case reflect.Int64:
+			fallthrough
+		case reflect.String:
+			// Valid type
+		default:
+			return &ConstraintError{
+				Message: fmt.Sprintf("invalid key type for map passed into 'any' type (%s); should be string or i64", reflectKind),
+			}
+		}
+		if firstReflectKind == reflect.Invalid { // First item
+			firstReflectKind = reflectKind
+		} else if firstReflectKind != reflectKind {
+			return &ConstraintError{
+				Message: fmt.Sprintf(
+					"mismatched key types in map passed into 'any' type: %s != %s",
+					firstReflectKind, reflectKind),
+			}
+		}
+
+		// Validate value
+		err := a.ValidateCompatibility(value)
+		if err != nil {
+			return &ConstraintError{
+				Message: fmt.Sprintf("validation error while validating map item item %q of map for 'any' type (%s)", key, err.Error()),
```

Comment:
Shouldn't this identify the expected and actual *type*?

Reply:
The sub-type's message is enough:
`
validation error while validating map item item "b" of map for 'any' type (Validation failed: unsupported data type for 'any' type: schema_test.someStruct)
`